### ISSUE TITLE
Filtering with diacritical marks

### ISSUE THREAD

**Issue by klobetime** [elix/elix#115]
It would be really useful to have an option on the components such as `FilterListBox` to ignore diacritical marks.  That would allow for a search of _para_ (for example) to return both _paralelogramo_ and _parábola_ in Spanish.

**Comment by JanMiksovsky** [comment]
Interesting.

It looks like we could use `localeCompare` to do the comparison. That takes the same `options` described for [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator).

If we used `sensitivity: base` (and maybe `ignorePunctuation: true`, while we're at it), I think we'd get what you want. Let me know if that looks correct to you.

**Comment by klobetime** [comment]
Yes, that looks exactly right!  The punctuation thing is interesting as well; that would take care of contractions (cant = can't) which is sometimes annoying as well.Thanks!

**Comment by JanMiksovsky** [comment]
Before I tackle this: do *you* have any interest in implementing this fix? I'm always looking for people to help out, and would guide you through it if you're interested. This could start simply, e.g., with writing a unit test for this that fails.

**Comment by JanMiksovsky** [comment]
Progress report: I'll be checking in a fix for this.

Like most international issues, this turns out to be more complicated than it first looks. The `Intl.Collator` object looks very promising for this situation — but unfortunately appears only used in _comparing_ strings. What `FilterListBox` does to _search_ strings.

I did find a helpful string normalization snippet that removes accents which seems to work. However, it's not strictly speaking correct. In some languages, accents matter. Example: ä and a have the same base letter in German, but in Swedish those are separate base letters. A Swedish user may therefore be disappointed/annoyed if those letters are treated the same for filtering purposes.

That said, removing accents seems more helpful in most cases, so I've gone ahead with that as a basis for a fix. I've also exposed a method on the component that a dev can override to customize the matching behavior for specific situations/locales.
